kwave/__init__.py

```diff
@@ -102,7 +110,18 @@ def download_binaries(system_os: str, bin_type: str):
         os.makedirs(binary_path, exist_ok=True)
 
         # Download the binary file
-        urllib.request.urlretrieve(url, os.path.join(binary_path, filename))
+        try:
+            urllib.request.urlretrieve(url, os.path.join(binary_path, filename))
+        except TimeoutError:
+            logging.log(logging.WARN, f"Download of {filename} timed out. "
+                                       "This can be due to slow internet connection. "
+                                       "Going to remove partially downloaded file to avoid future problems.")
```

Comment:
```suggestion
        except TimeoutError:
            logging.log(logging.WARN, f"Download of {filename} timed out. "
                                       "This can be due to slow internet connection. "
                                       "Partially downloaded files will be removed.")
```